Crates_Server_side/services/level.js

```diff
@@ -0,0 +1,163 @@
+const uuidv1 = require('uuid/v1');
+const s3Service = require('./s3Service');
+
+let listAllObjectKeysFromS3Bucket = async (prefix) => {
+    let isTruncated = true;
+    let marker;
+    let allObjectKeysList = [];
+    while(isTruncated) {
+        const response = await s3Service.listObjects(marker, prefix);
+        response.Contents.forEach(item => {
+            allObjectKeysList.push(item.Key);
+        });
+        isTruncated = response.IsTruncated;
+        if (isTruncated) {
+            marker = response.Contents.slice(-1)[0].Key;
+        }
+    }
+    return allObjectKeysList;
+}
+
+let listAllLevelKeys = (allObjectKeysList) => {
+    return allObjectKeysList.filter( objectKey => {
+        return objectKey[objectKey.length - 1] !== "/"; });
+}
+
+exports.saveLevel = (levelMetadataJSON, userId) => {
+    levelMetadataJSON.updatedAt = Date.now();
+    if (!levelMetadataJSON.createdAt) {
+        levelMetadataJSON.createdAt = levelMetadataJSON.updatedAt;
+    }
+    if (!levelMetadataJSON.id) {
+        levelMetadataJSON.id = uuidv1();
+        levelMetadataJSON.createdBy = userId;
+    }
+    return new Promise((resolve, reject) => {
+        s3Service.upload(levelMetadataJSON, (error, data) => {
+            if (error) {
+                let response = {
+                    err: true,
+                    data: error
+                };
+                reject(response);
+            }
+            else {
```

Comment:
It is generally a good practice to catch an error in an `if` block and return, putting the rest of the code in the outer block, rather than using an `if...else` block that will indent a large block of code inside. This becomes a bigger problem when you have nested `if` conditions

Reply:
Yes, I understood what you have said. I also had the same habit. I read somewhere that, it is better to have the code classified into `if` and `else` blocks, to improve code clarity and understandability. I have no issues to implement the way you have said.